internal/httpclient/httpclient.go

```diff
@@ -0,0 +1,40 @@
+// Package httpclient contains the default HTTP client. This client is used
+// for normal operations an honours the HTTP_PROXY environment variable.
+package httpclient
+
+import (
+	"net/http"
+
+	"github.com/ooni/probe-engine/internal/dialer"
+	"github.com/ooni/probe-engine/internal/httptransport"
+	"github.com/ooni/probe-engine/internal/resolver"
+	"github.com/ooni/probe-engine/internal/tlsdialer"
+)
+
+// Logger is the logger interface assumed by this package
+type Logger interface {
+	Debugf(format string, v ...interface{})
+}
+
+// New creates a new HTTP client.
+func New(logger Logger) *http.Client {
```

Comment:
This should probably take in input a `Dialer`, so we can create a client with undetermined real `Dialer` by using the `Service` and we can hence control a bunch of stuff that happens, as well as wrap the `Conn` before passing it back to the HTTP code.